backend/views/retrieve_test_view.py

```diff
@@ -0,0 +1,135 @@
+    return retrieve_response_from_name_date(request, query_date_time, request_type)
+
+# Wrap the json in a Response; this makes testing easier
+
+
+def retrieve_response_from_name_date(request, query_date_time, request_type):
+    # get the test_name from the parameter
+    test_name = request.query_params.get('test_name', None)
+    return Response(retrieve_json_from_name_date(test_name, query_date_time, request_type))
+
+
+def retrieve_json_from_name_date(test_name, query_date_time, request_type):
+    # if there is no test_name prop, then return an error
+    if test_name is None:
+        return {"error", "no 'test_name' parameter"}
+    # if there is a test_name, look it up
+    try:
+        test = Test.objects.get(test_name=test_name)
+    except Test.DoesNotExist:
+        return {"error", "no test with the given test_name"}
+    # get the associated item
+    item_id = test.item_id_id
+    if item_id is None:
+        return {"error", "no item id in test"}
+
+    # Ensure that the item with the same pk,
+    # is active (where the current date is between the from and to date)
+    try:
+        Item.objects.get(
+            pk=item_id, date_from__lte=query_date_time, date_to__gt=query_date_time)
+    except Item.DoesNotExist:
+        # if the above fails, try to get an item with the same id, after the from date
+        # and where the to date is null
+        try:
+            Item.objects.get(
+                pk=item_id, date_from__lte=query_date_time, date_to__isnull=True)
+        except Item.DoesNotExist:
+            return {"error", "no test item found"}
+
+    # get item text (visible test names)
+    en_id, fr_id = get_language_ids(query_date_time)
+
+    # get the user visible names, if they exist
+    try:
+        en_name = ItemText.objects.get(
+            item_id=item_id, language=en_id,
+            date_from__lte=query_date_time,
+            date_to__gt=query_date_time).text_detail
+    except ItemText.DoesNotExist:
+        try:
+            en_name = ItemText.objects.get(
+                item_id=item_id, language=en_id,
+                date_from__lte=query_date_time,
+                date_to__isnull=True).text_detail
+        except ItemText.DoesNotExist:
+            en_name = None
+
+    try:
+        fr_name = ItemText.objects.get(
+            item_id=item_id, language=fr_id,
+            date_from__lte=query_date_time,
+            date_to__gt=query_date_time).text_detail
+    except ItemText.DoesNotExist:
+        try:
+            fr_name = ItemText.objects.get(
+                item_id=item_id, language=fr_id,
+                date_from__lte=query_date_time,
+                date_to__isnull=True).text_detail
+        except ItemText.DoesNotExist:
+            fr_name = None
+
+    # if only looking for the meta data, then return
+    return_dict = {'test_internal_name': test.test_name,
+                   'test_en_name': en_name,
+                   'meta_test.test_fr_name': fr_name,

```

Comment:
Pre-pending these names with `meta_test.` seems like it might cause some confusion in the future, especially when retrieving the values in JS. Can we remove `meta_test` everywhere?

Reply:
typo on my part. Dropped in 13c788a (will drop in other branches too)